### ISSUE TITLE
nan in elbo

### ISSUE THREAD

**Issue by luiarthur** [luiarthur/CytoPy#2]
Getting nan's in elbo. I'm guessing it is related to the imputed y values.

I have preprocessed data to exclude cells containing expressions < -6.0

Perhaps somethings else needs to be done.

**Comment by luiarthur** [comment]
Figured out the problem. Related to [this][1].

I was having nan's a lot. And even stranger things. e.g., no nan's in my personal laptop, but nan's very quickly on the server... ???

Basically, using a smaller learning weight, **and** using double precision (instead of single) fixed the problem. My guess is the gradients are very unstable. It helps to use a larger learning rate for only a few iterations, then use a smaller learning rate for a much longer period (or until convergence).

[1]: https://discuss.pytorch.org/t/why-double-precision-training-sometimes-performs-much-better/31194
